managed/services/management/azure_database.go

```diff
@@ -98,45 +97,38 @@ type AzureDatabaseInstanceData struct {
 	Zones         string                 `json:"zones"`
 }
 
-func (s *AzureDatabaseService) getAzureClient(req *azurev1beta1.DiscoverAzureDatabaseRequest) (*resourcegraph.BaseClient, error) {
-	authSettings := auth.EnvironmentSettings{
-		Values: map[string]string{
-			auth.ClientID:       req.AzureClientId,
-			auth.ClientSecret:   req.AzureClientSecret,
-			auth.SubscriptionID: req.AzureSubscriptionId,
-			auth.TenantID:       req.AzureTenantId,
-			auth.Resource:       azure.PublicCloud.ResourceManagerEndpoint,
-		},
-		Environment: azure.PublicCloud,
+func (s *AzureDatabaseService) getAzureClient(req *azurev1beta1.DiscoverAzureDatabaseRequest) (*armresourcegraph.Client, error) {
+	credential, err := azidentity.NewClientSecretCredential(req.AzureTenantId, req.AzureClientId, req.AzureClientSecret, nil)
+	if err != nil {
+		return nil, err
 	}
 
 	// Create and authorize a ResourceGraph client
-	client := resourcegraph.New()
-	authorizer, err := authSettings.GetAuthorizer()
+	client, err := armresourcegraph.NewClient(credential, nil)
 	if err != nil {
 		return nil, err
 	}
 
-	client.Authorizer = authorizer
-	return &client, nil
+	return client, nil
 }
 
 func (s *AzureDatabaseService) fetchAzureDatabaseInstancesData(
 	ctx context.Context,
 	req *azurev1beta1.DiscoverAzureDatabaseRequest,
-	client *resourcegraph.BaseClient,
+	client *armresourcegraph.Client,
 ) ([]AzureDatabaseInstanceData, error) {
 	query := azureDatabaseResourceQuery
-	request := resourcegraph.QueryRequest{
-		Subscriptions: &[]string{req.AzureSubscriptionId},
+	resultFormat := "objectArray"
+	request := armresourcegraph.QueryRequest{
+		Subscriptions: []*string{&req.AzureSubscriptionId},
 		Query:         &query,
-		Options: &resourcegraph.QueryRequestOptions{
-			ResultFormat: "objectArray",
+		Options: &armresourcegraph.QueryRequestOptions{
+			ResultFormat: (*armresourcegraph.ResultFormat)(&resultFormat),
```

Comment:
And here we can remove converting then
```suggestion
			ResultFormat: &resultFormat,
```